ale_linters/dockerfile/hadolint.vim

```diff
@@ -7,26 +7,46 @@ call ale#Set('dockerfile_hadolint_docker_image', 'lukasmartinelli/hadolint')
 function! ale_linters#dockerfile#hadolint#Handle(buffer, lines) abort
     " Matches patterns line the following:
     "
-    " stdin:19: F: Pipe chain should start with a raw value.
-    let l:pattern = '\v^/dev/stdin:?(\d+)? (\S+) (.+)$'
+    " /dev/stdin:19 DL3001 Pipe chain should start with a raw value.
+    " /dev/stdin:19:3 unexpected thing
+    let l:pattern = '\v^/dev/stdin:(\d+):?(\d+)? ((DL|SC)(\d+) )?(.+)$'
     let l:output = []
 
     for l:match in ale#util#GetMatches(a:lines, l:pattern)
         let l:lnum = 0
+        let l:colnum = 0
 
         if l:match[1] isnot# ''
             let l:lnum = l:match[1] + 0
         endif
 
+        if l:match[2] isnot# ''
+            let l:colnum = l:match[2] + 0
+        endif
+
         let l:type = 'W'
-        let l:text = l:match[3]
+        let l:text = l:match[6]
+        let l:detail = l:match[6]
+        let l:domain = 'https://github.com/hadolint/hadolint/wiki/'
+
+        if l:match[4] is# 'SC'
+            let l:domain = 'https://github.com/koalaman/shellcheck/wiki/'
+        endif
```

Comment:
Could you add an option for this, and leave it off by default? This will make the messages too long for some users.

Reply:
This is only shown in the error detail, I think that full error messages are more often than not longer than this, aren't they?

Reply:
Ah, I see. That's okay then.